fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/AgilityContextTest.java

```diff
@@ -414,6 +418,39 @@ void testAgilityContextAtomicAttribute(boolean useAgile) {
         }
     }
 
+    @Test
+    void testAgilityContextRecoveryPath() {
+        AtomicInteger refInt = new AtomicInteger(0);
+        try (FDBRecordContext context = openContext()) {
+            final AgilityContext agilityContext = getAgilityContext(context, true);
+            final Subspace subspace = path.toSubspace(context);
+            byte[] key = subspace.pack(Tuple.from(2023, 3));
+            byte[] val = Tuple.from(800, "Green eggs and ham", 0).pack();
+            agilityContext.set(key, val);
+            agilityContext.abortAndClose();
+            IntStream range = IntStream.rangeClosed(1, 10);
+            range.parallel().forEach(i -> {
+                if (i == 7) {
+                    agilityContext.applyInRecoveryPath(aContext -> {
+                        Assertions.assertNotNull(aContext);
+                        refInt.addAndGet(i);
```

Comment:
It seems like this should write something to the database, and afterwards we should assert that it was written with the correct value.

Reply:
✅